### PR TITLE
Document platform-specific differences for `std::process::exit`.

### PR DESCRIPTION
**Pull request by frewsxcv**
Fixes https://github.com/rust-lang/rust/issues/35046.

### CODE REVIEW

File: src/libstd/process.rs [hunk 1]
```diff
@@ -828,13 +828,33 @@ impl Child {
...
 /// to run.
 ///
+/// ## Platform-specific behavior
+///
+/// Despite `code` being an `i32`, some platforms take into account only a
+/// certain range of these bits. On macOS and Linux, `exit`'s underlying system
+/// function only considers the lower eight bits of `code`. For example, on
+/// these platforms, calling `exit` with values `0x00_00` and `0x0f_00` will be
+/// handled identically. Other platforms, like Windows, will take into account
+/// all 32 bits of `code`.
```
Comment (frewsxcv) [inline src/libstd/process.rs]: I was going to add a sentence at the end like:

> For specific information about your platform, please check \<insert resource here\>

But I wasn't sure what resource to put there. `man 3 exit`?  "your systems 'exit' function"?

¯\\\_(ツ)_/¯

Comment (alexcrichton) [inline src/libstd/process.rs]: I tend to like to cite the [opengroup docs](http://pubs.opengroup.org/onlinepubs/000095399/functions/exit.html) as they've always seemed "extra official". Maybe a link to that could be added here?

**Comment by alexcrichton** [comment]
Thanks @frewsxcv! In retrospect we should have probably made this argument a `u8`, but alas.

File: src/libstd/process.rs (hunk 1 above)
Comment (frewsxcv) [inline src/libstd/process.rs]: It looks like your opengroup link says `exit` only takes into consideration the least significant eight bits. I have hesitations about *just* citing that as it doesn't match the behavior on Windows which takes into account *all* 32 bits with no truncation (as per [this comment](https://github.com/rust-lang/rust/issues/35046#issuecomment-235484403) by @retep998).

Comment (alexcrichton) [inline src/libstd/process.rs]: Oh sorry to be clear that opengroup page is a Unix-only thing, which is basically just explaining why only 8 bits are transmitted to the parent on Unix.

File: src/libstd/process.rs [hunk 2]
```diff
@@ -828,13 +828,33 @@ impl Child {
...
+///
 /// # Examples
 ///
 /// ```
 /// use std::process;
 ///
 /// process::exit(0);
 /// ```
+///
+/// Due to [platform-specific behavior], the exit code for this example will be
```
Comment (nagisa) [inline src/libstd/process.rs]: This is a great example.

File: src/libstd/process.rs [hunk 3]
```diff
@@ -828,13 +828,30 @@ impl Child {
...
+/// `0` on Linux, but `256` on Windows:
+///
+/// ```
```
Comment (frewsxcv) [inline src/libstd/process.rs]: This is going to fail on Windows, because it's a non-zero return code. So I'm going to mark as `no_run`.

**Pull request merged**